### PR TITLE
Service account test

### PR DESCRIPTION
**Pull request by mmusil**

### CODE REVIEW

File: lib/sa.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+export const createServiceAccount = async function (page: Page, name: string) {
+    await page.locator('button', { hasText: 'Create service account' }).click();
+
+    await expect(page.getByText('Create a service account', {exact: true})).toHaveCount(1);
+
+    await page.waitForSelector('[role=progressbar]', { state: 'detached' });
+
+    await page.getByLabel('Short description *').fill(name);
+
+    await page.getByTestId('modalCreateServiceAccount-buttonSubmit').click();
```
Comment (mmusil) [inline lib/sa.ts:20]: @Frawless I know you and @kornys removed probably all data-testid locators, I tried to find another reliable locator for this one, but no other except maybe for `await page.getByText('Create', {exact: true}).click();` would work.
If I used the simple `page.locator('button', { hasText: 'Create' })`, there is no exact match option, so it returns even the 'Create service account' button hidden behind the modal.

File: lib/sa.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    await page.getByRole('link', { name: 'Application and Data Services' }).click();
+    // }
+    await page.getByRole('link', { name: 'Service Accounts' }).click();
+    await expect(page.getByRole('heading', { name: 'Service Accounts' })).toHaveCount(1);
+};
+
+export const createServiceAccount = async function (page: Page, name: string) {
+    await page.locator('button', { hasText: 'Create service account' }).click();
+
+    await expect(page.getByText('Create a service account', {exact: true})).toHaveCount(1);
```
Comment (Frawless) [inline lib/sa.ts:14]: can you use something like `await page.locator('button', { hasText: 'Create service account' }).click();` ? From my POV it should work

File: lib/sa.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    // check for the service account to have been created
+    const table = await page.locator('[aria-label="Service account list"]');
+    await expect(table.getByText(name)).toHaveCount(1);
+
+    return {clientID: clientID, clientSecret: clientSecret}
+};
+
+export const deleteServiceAccount = async function (page: Page, name: string) {
+    console.log("Deleting service account " + name);
```
Comment (andreaTP) [inline lib/sa.ts:43]: [nitpick] this should not be necessary anymore

File: lib/sa.ts (hunk 1 above)
Comment (Frawless) [reply lib/sa.ts:20]: can you try `page.locator('.button :text-is("Create")')` ?

File: tests/sa.spec.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  const elements = async () => {
+    return await (
+      await page.locator(`tr >> td[data-label="Description"]`).elementHandles()
+    ).length;
+  };
+
+  while ((await elements()) > 0) {
+    const el = await page.locator(`tr >> td[data-label="Description"]`).nth(0);
+    console.log('element from beforeEach: ' + el);
```
Comment (andreaTP) [inline tests/sa.spec.ts:24]: [nitpick] the `console.log` can probably be removed now.

File: tests/sa.spec.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    const accountID = await el.textContent();
+    console.log('Account to be deleted: ' + accountID);
+    await deleteServiceAccount(page, accountID);
+  }
+});
+
+// test_5sa.py test_sa_create
+test('test service account creation', async ({ page }) => {
+  const testServiceAccountName = `test-service-account-${config.sessionID}`;
```
Comment (andreaTP) [inline tests/sa.spec.ts:33]: [nitpick] wanna move this line to the top of the file and reuse it?

**Review by andreaTP — APPROVED** [review]
a few nitpicks but is already good enough to merge 👍  LGTM!

File: lib/sa.ts (hunk 1 above)
Comment (mmusil) [reply lib/sa.ts:20]: Strangely enough, removing the dot before the button and the space after it (`'button:text-is("Create")'`) did the trick. The space is in the documentation, so I don't really get it.
Or perhaps using `.getByText` with regular expressions  `'/^Create$/'` could do.

Comment (Frawless) [reply lib/sa.ts:20]: Interesting...I am not UI testing guru so I will let decision up to you :)

**Review by Frawless — APPROVED** [review]

File: lib/sa.ts (hunk 1 above)
Comment (Frawless) [reply lib/sa.ts:20]: I think as long as the tests will be stable we are fine

Comment (andreaTP) [reply lib/sa.ts:20]: is the commit missing?

**Pull request merged**